### PR TITLE
tg-can-693

### PR DESCRIPTION
**Pull request by nalwa**
This PR has implementation for API can-693 and can-694

@akuks  please review and merge PR.

### CODE REVIEW

File: app/Helpers/TopicSupport.php [hunk 1]
```diff
@@ -591,5 +591,146 @@ public static function SendEmailToSubscribersAndSupporters($data)
...
+
+    /**
+     * 
+     */
+    public static function checkIfDelegatorSupporter($topicNum, $campNum, $nickNames)
+    {
+        $returnData = [];
+        $supportedCamps = [];
+        $delegatedSupport = Support::getDelgatedSupportInTopic($topicNum,$nickNames);                
+        if ($delegatedSupport->count() > 0) {
```
Comment (akuks) [inline app/Helpers/TopicSupport.php:627]: I think we can simple leave it to `if ($delegatedSupport->count() )`

File: app/Helpers/TopicSupport.php [hunk 2]
```diff
@@ -591,5 +591,146 @@ public static function SendEmailToSubscribersAndSupporters($data)
...
+        $parentSupport = Camp::validateParentsupport($topicNum, $campNum, $nickNames);
+
+        $filter = Camp::getLiveCampFilter($topicNum, $campNum);
+        $onecamp = self::getLiveCamp($filter);        
+
+        if(!empty($parentSupport)){
+
+            if ($parentSupport === "notlive") { 
+                               
+                $returnData['warning'] =  "You can not submit your support to this camp as its not live yet.";
```
Comment (akuks) [inline app/Helpers/TopicSupport.php:665]: May be from a common message file

File: app/Helpers/TopicSupport.php [hunk 3]
```diff
@@ -591,5 +591,146 @@ public static function SendEmailToSubscribersAndSupporters($data)
...
+    public static function checkIfSupportswitchToChild($topicNum, $campNum, $nickNames)
+    {
+        $returnData = [];
+        $as_of_time = time();
+        $parentSupport = Camp::validateParentsupport($topicNum, $campNum, $nickNames);
+
+        $filter = Camp::getLiveCampFilter($topicNum, $campNum);
+        $onecamp = self::getLiveCamp($filter);        
+
+        if(!empty($parentSupport)){
```
Comment (akuks) [inline app/Helpers/TopicSupport.php:661]: Can we refactor this? There is too much nested if else.

File: app/Models/Camp.php [hunk 4]
```diff
@@ -427,4 +427,55 @@ public static function getDirectCampSupporterIds($topicid, $campnum)
...
+        $oneCamp = self::getLiveCamp($filter);        
+
+        if ($oneCamp->count() <= 0) {
+            return 'notlive';
+        }       
+
+        $parentcamps = self::getAllParent($oneCamp);
+        $mysupports = Support::where('topic_num', $topicNum)->whereIn('camp_num', $parentcamps)->whereIn('nick_name_id', $userNicknames)->where('end', '=', 0)->orderBy('support_order', 'ASC')->get();
+        
+        if (count($mysupports))
```
Comment (akuks) [inline app/Models/Camp.php:443]: probably a ternary operator will reduce the complexity

**Review by akuks — CHANGES_REQUESTED** [review]

**Comment by nalwa** [comment]
@akuks  I have done changes requested and suggested by you. please review once again.

**Review by akuks — APPROVED** [review]

**Pull request merged**